Lacrit/assignment4/islands.cpp

```diff
@@ -0,0 +1,68 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+ 
+const int rows = 5;
+const int cols = 5;
+
+// A function to check if a given cell (row, col) can be included in DFS
+int isValid(int mat[rows][cols], int row, int col, bool visited[rows][cols])
+{
+    // row number is in range, column number is in range and value is 1 
+    // and not yet visited
+    return (row >= 0) && (row < rows) &&     
+           (col >= 0) && (col < cols) &&      
+           (mat[row][col] && !visited[row][col]); 
+}
+ 
+// check all 8 neighbours
+void DFS(int mat[rows][cols], int row, int col, bool visited[rows][cols])
+{
+    // These arrays are used to get row and column numbers of 8 neighbours 
```

Comment:
Note that the problem statement specifies that we do **not** take diagonal connections.